src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -305,7 +305,25 @@ public override void VisitEqualsValueClause(EqualsValueClauseSyntax node)
 
         public override void VisitAttribute(AttributeSyntax node)
         {
-            var attrBinder = new ExpressionVariableBinder(node, _enclosing);
+            Binder attrBinder = null;
+            if (node.Parent?.Parent is { } attributedDeclaration)
+            {
+                if (attributedDeclaration is LocalFunctionStatementSyntax localFunctionSyntax)
+                {
+                    var localFunction = FindLocalFunction(localFunctionSyntax, _enclosing);
+                    attrBinder = new ContextualAttributeBinder(_enclosing, localFunction);
+                }
+                else
+                {
+                    var model = _containingMemberOrLambda.DeclaringCompilation.GetSemanticModel(node.SyntaxTree);
```

Comment:
>var model = _containingMemberOrLambda.DeclaringCompilation.GetSemanticModel(node.SyntaxTree); [](http://example.com/codeflow?start=20&length=93)

Changes like that in this component feel wrong. In general, I am surprised that we need to make any changes outside of semantic model and IOperation factory.